File: 백준/Silver/1138. 한 줄로 서기/한 줄로 서기.py
import sys

def solution(n, arr):
    res = [0] * n

    for i in range(n):
        for j in range(n):
            if arr[i] == 0 and res[j] == 0:
                res[j] = i + 1
                break
            elif res[j] == 0:
                arr[i] -= 1

    return res

if __name__ == "__main__":
    n = int(input())
    arr = list(map(int, sys.stdin.readline().rstrip().split()))

    for x in solution(n, arr):
        print(x, end = ' ')

File: 백준/Silver/1138. 한 줄로 서기/한 줄로 서기.java
import java.util.Scanner;

class Main {
    public static int[] solution(int n, int[] arr) {
        int[] res = new int[n];

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                if (arr[i] == 0 && res[j] == 0) {
                    res[j] = i + 1;
                    break;
                } else if (res[j] == 0) {
                    arr[i] -= 1;
                }
            }
        }

        return res;
    }

    public static void main(String[] args) {
        Scanner kb = new Scanner(System.in);

        int n = kb.nextInt();
        int[] arr = new int[n];

        for (int i = 0; i < n; i++) {
            arr[i] = kb.nextInt();
        }

        for (int x : solution(n, arr)) {
            System.out.print(x + " ");
        }
    }
}